src/user/user.controller.ts

```diff
@@ -0,0 +1,55 @@
+import { Router, Request, Response, NextFunction } from 'express';
+import { IUser } from './user.interface';
+import { IController } from '../interfaces/controller.interface';
+import { IHelperArgs } from '../interfaces/helpers.interface';
+import InMemoryDatabase from '../database';
+import * as helpers from './user.helpers';
+import DbAdapter from '../database/db.adapter';
+import { loader } from '../database/loader';
+
+class UserController implements IController {
+    public path = '/users';
+    public router = Router();
+
+    private usersDb = new InMemoryDatabase<IUser>();
+
+    constructor() {
+        this.initializeRoutes();
+        loader('src/database/data.json', new DbAdapter<IUser, InMemoryDatabase<IUser>>(this.usersDb));
```

Comment:
Consider extracting hard-coded strings into separate constants at the top of the file.
In the production case filenames would be dynamic and paths would probably be passed down to the script from the outside (env variable or CLI argument) to make the whole thing reusable.